### PR TITLE
P.flis/diffuse shader

### PR DESCRIPTION
**Pull request by p-flis**

### CODE REVIEW

File: RayTracer/Resources/Shaders/shader.frag [hunk 1]
```diff
@@ -20,5 +34,29 @@ void main()
...
     }
-    FragColor = vec4(ambientLight, 1.0) * color;
+    if (isLight != 0)
+    {
+        FragColor = color;
+        return;
+    }
+    vec3 normal = normalize(Normal);
+    vec3 lights = vec3(0.0, 0.0, 0.0);
+    if (lightsCount != 0)
```
Comment (karolk98) [inline RayTracer/Resources/Shaders/shader.frag:44]: This does nothing

File: RayTracer/Resources/Shaders/shader.frag [hunk 2]
```diff
@@ -20,5 +34,29 @@ void main()
     else {
         color = vec4(materialColor, 1.0);
     }
-    FragColor = vec4(ambientLight, 1.0) * color;
+    if (isLight != 0)
```
Comment (karolk98) [inline RayTracer/Resources/Shaders/shader.frag:37]:
```suggestion
    if (isLight == 1)
```

**Review by karolk98 — APPROVED** [review]

File: RayTracer/Source/Materials/Emissive.cs [hunk 3]
```diff
@@ -1,22 +1,37 @@
...
+        public Color AverageColor => _averageColor;
 
+        private ITexture _emit;
+        private Color _averageColor;
+        
         public Emissive(ITexture emit)
         {
             _emit = emit;
+            int samples = 10000;
+            var sampler = new ThreadSafeSampler<Vector2>(Vec2Sampling.Random, samples);
```
Comment (karolk98) [inline RayTracer/Source/Materials/Emissive.cs:21]: This does not have to be thread safe

File: RayTracer/Resources/Shaders/shader.frag (hunk 2 above)
Comment (tomasz-herman) [reply RayTracer/Resources/Shaders/shader.frag:37]: it's good as is

File: RayTracer/Source/Materials/Emissive.cs (hunk 3 above)
Comment (tomasz-herman) [reply RayTracer/Source/Materials/Emissive.cs:21]: w/e

**Review by tomasz-herman — APPROVED** [review]

**Pull request merged**